### PR TITLE
[VGNWB-314] Add password reset views

### PR DESCRIPTION
**Pull request by nowke**
Views for

- Forgot password - Sends email in the console
- Reset password

### CODE REVIEW

**Comment by tjquinn1** [comment]
Resolve conflicts and is there a client a ticket for this as well?

**Comment by nowke** [comment]
Corresponding ticket - https://diging.atlassian.net/browse/VGNWB-314

**Comment by tjquinn1** [comment]
Resolve conflicts

File: accounts/views.py [hunk 1]
```diff
@@ -73,6 +76,81 @@ def citesphere_token(request):
...
+        username = request.data.get('username', None)
+        if username:
+            try:
+                user = VogonUser.objects.get(username=username)
+            except VogonUser.DoesNotExist:
+                return Response({ "success": False, "message": "User not found" }, status=404)
+            token = jwt.encode({ 
+                'exp': arrow.utcnow().shift(days=1).timestamp,
+                'username': user.username,
+            }, settings.SECRET_KEY, algorithm='HS256').decode('UTF-8')
```
Comment (tjquinn1) [inline accounts/views.py:90]: why do you encode the secret key here?

File: accounts/views.py [hunk 2]
```diff
@@ -73,6 +76,81 @@ def citesphere_token(request):
...
+            reset_token = ResetToken(user=user, token=token)
+            reset_token.save()
+
+            # Email token
+            send_mail(
+                'Vogon password reset',
+                f'Reset link: http://localhost:8080/reset-password/{token}',
```
Comment (tjquinn1) [inline accounts/views.py]: lets make the reset link and the vogon email be configurable from the settings file

File: accounts/views.py [hunk 3]
```diff
@@ -73,6 +76,81 @@ def citesphere_token(request):
...
+                'test@vogonweb.com',
+                [user.email],
+                fail_silently=True,
```
Comment (tjquinn1) [inline accounts/views.py]: do we want this to fail silently?

File: accounts/views.py [hunk 4]
```diff
@@ -73,6 +76,81 @@ def citesphere_token(request):
...
+            )
+
+            return Response({ "success": True })
+        return Response({ "success": False, "message": "Username not specified" }, status=400)
+
+class ResetPasswordView(APIView):
+    def post(self, request):
+        username = request.data.get('username', None)
```
Comment (tjquinn1) [inline accounts/views.py]: I think we should use a serializer for this. That way we can validate the data

File: vogon/settings.py [hunk 5]
```diff
@@ -62,6 +62,7 @@
     "django.middleware.security.SecurityMiddleware",
 )
 
+EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
```
Comment (tjquinn1) [inline vogon/settings.py]: lets make this an env variables

**Review by tjquinn1 — CHANGES_REQUESTED** [review]

File: accounts/views.py (hunk 1 above)
Comment (nowke) [reply accounts/views.py:90]: JWT uses secret key to encode the contents of the dict.

File: accounts/views.py (hunk 3 above)
Comment (nowke) [reply accounts/views.py]: Fixed

File: accounts/views.py (hunk 4 above)
Comment (nowke) [reply accounts/views.py]: Done

File: vogon/settings.py (hunk 5 above)
Comment (nowke) [reply vogon/settings.py]: Done

**Comment by tjquinn1** [comment]
Make it so, Jenkins.

**Pull request merged**